Userland/Services/FileOperation/main.cpp

```diff
@@ -255,30 +224,18 @@ int execute_work_items(Vector<WorkItem> const& items)
             outln("PROGRESS {} {} {} {} {} {} {}", i, items.size(), executed_work_bytes, total_work_bytes, item_done, item.size, item.source);
         };
 
-        auto copy_file = [&](String const& source, String const& destination) {
-            auto source_file_or_error = Core::File::open(source, Core::OpenMode::ReadOnly);
-            if (source_file_or_error.is_error()) {
-                report_warning(String::formatted("Failed to open {} for reading: {}", source, source_file_or_error.error()));
-                return false;
-            }
+        auto copy_file = [&](String const& source, String const& destination) -> ErrorOr<int> {
+            auto source_file = TRY(Core::File::open(source, Core::OpenMode::ReadOnly));
             // FIXME: When the file already exists, let the user choose the next action instead of renaming it by default.
-            auto destination_file_or_error = open_destination_file(destination);
-            if (destination_file_or_error.is_error()) {
-                report_warning(String::formatted("Failed to open {} for write: {}", destination, destination_file_or_error.error()));
-                return false;
-            }
-            auto& source_file = *source_file_or_error.value();
-            auto& destination_file = *destination_file_or_error.value();
+            auto destination_file = TRY(open_destination_file(destination));
 
             while (true) {
                 print_progress();
-                auto buffer = source_file.read(65536);
+                auto buffer = source_file->read(65536);
                 if (buffer.is_empty())
                     break;
-                if (!destination_file.write(buffer)) {
-                    report_warning(String::formatted("Failed to write to destination file: {}", destination_file.error_string()));
-                    return false;
-                }
+                if (!destination_file->write(buffer))
+                    return Error::from_string_literal(destination_file->error_string());
```

Comment:
What does an error string look like here? Somehow keeping the information that we specifically failed at writing the destination file would likely be useful, but we'd likely run into the formatting issues again, as described above.

Reply:
Changed back to original.